astropy/io/fits/tests/test_image.py

```diff
@@ -1046,11 +1043,10 @@ def test_invalid_blank(self):
         data = np.arange(100, dtype=np.float64)
         hdu = fits.PrimaryHDU(data)
         hdu.header["BLANK"] = "nan"
-        with pytest.warns(
-            fits.verify.VerifyWarning,
-            match=r"Invalid value for 'BLANK' keyword in header: 'nan'",
-        ):
+
+        with pytest.warns(fits.verify.VerifyWarning) as w:
             hdu.writeto(self.temp("test.fits"))
+        assert "Invalid 'BLANK' keyword in header" in str(w[1].message)
```

Comment:
The old text failed the test. Not sure if I translated this correctly. @saimn , please double check.

Reply:
OK nvm, index error. Ignore this.